src/main/java/com/dev/cinema/dao/UserDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.dev.cinema.dao;
+
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.Optional;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add user " + user + " in to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> userQuery =
+                    session.createQuery("from User where email = :email", User.class);
+            userQuery.setParameter("email", email);
+            return userQuery.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get user with email " + email, e);
+        }
```

Comment:
No need in catching the exception, because if you receive null you'll simply return the empty Optional

Reply:
fixed

Reply:
> No need in catching the exception, because if you receive null you'll simply return the empty Optional

You won't get exception here in case of null, but true it's not necessary to catch exception while read operations